dali/operators/reader/loader/numpy_loader.h

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_LOADER_NUMPY_LOADER_H_
+#define DALI_OPERATORS_READER_LOADER_NUMPY_LOADER_H_
+
+#include <dirent.h>
+#include <sys/stat.h>
+#include <errno.h>
+
+#include <fstream>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+#include <algorithm>
+#include <regex>
+#include <memory>
+
+#include "dali/core/common.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/operators/reader/loader/file_loader.h"
+#include "dali/util/file.h"
+
+
+namespace dali {
+TypeInfo TypeFromNumpyStr(const std::string &format);
+
+class NumpyParseTarget{
+ public:
+  std::vector<int64_t> shape;
+  TypeInfo type_info;
+  bool fortran_order;
+
+  size_t size() {
+    if (shape.empty()) return 0;
```

Comment:
In numpy an empty shape has a size equal to 1, not 0.
See
```Python
>>> x = numpy.float32(42)
>>> x.shape
()
>>> x.size
1
```

Reply:
I know, but I set it to 1 so it will produce an output tensor with a single element. It is easier to deal with it that way when hooking it into DALI. np.array_equal will still return true if you compare a 1-element array to a singleton. Or how would you handle singleton and DALI tensors?

Reply:
Do you ever have a situation where the shape would be empty? If not, then it doesn't matter how you handle empty shapes, because there are none. At present, DALI's volume function will give you 0 for an empty shape (although we plan to change it and allow proper scalars with empty shape). Basically, the entire body of this function could be replaced by:
```C++
return volume(shape);
```
and it would produce exactly the same behavior.

Reply:
Now I see what you mean. Yes, I can fix that. Empty shape does not mean (), because () is mapped to (1) in the reader so that it can read singletons. Empty shape means someone used this function in an illegal way. But I can remove that and replace by volume. The question is that if I should use the TensorShape instead, then I won't have to do that. But I had some difficulties with TensorShape<>. How can I resize it dynamically?

Reply:
fixed